features/step_definitions/navigation_steps.rb

```diff
@@ -1,3 +1,12 @@
 Given(/^I am on the landing page$/) do
   visit root_path
-end
\ No newline at end of file
+end
+
+Given(/^I am on the list applications page$/) do
+  visit memberships_path
+end
+
+Then(/^I should be on "([^"]*)" page$/) do |name|
+  @membership = MembershipApplication.find_by(company_name: name)
```

Comment:
A local variable will do. No need to store the MembershipApplication in an instance variable.